### PR TITLE
Bugfixes for 5.0 beta

### PR DESCRIPTION
**Pull request by janst97**

### CODE REVIEW

File: Monal/Classes/MLUploadQueueItem.m [hunk 1]
```diff
@@ -39,32 +39,37 @@ -(NSURL*) getURL
 
 -(id) initWithImage:(UIImage*) image {
     assert(image != nil);
-    assert(self = [super init]);
-    _type = UPLOAD_QUEUE_TYPE_RAW_IMAGE;
-    self.image = image;
-    self.url = nil;
+    if(self = [super init])
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MLUploadQueueItem.m]: [super init]mus ALWAYS be called and it has to be the first line in the init method (according to apple docs)

File: Monal/Classes/MLUploadQueueItem.m [hunk 2]
```diff
@@ -39,32 +39,37 @@ -(NSURL*) getURL
...
 
 -(id) initWithImage:(UIImage*) image imageUrl:(NSURL *)url
 {
     assert(image != nil);
     assert(url != nil);
-    assert(self = [super init]);
-    _type = UPLOAD_QUEUE_TYPE_IMAGE_WITH_URL;
-    self.image = image;
-    self.url = url;
+    if(self = [super init])
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MLUploadQueueItem.m]: same here

File: Monal/Classes/MLUploadQueueItem.m (hunk 1 above)
Comment (tmolitor-stud-tu) [reply Monal/Classes/MLUploadQueueItem.m]: why that if btw?

File: Monal/Classes/MLUploadQueueItem.m [hunk 3]
```diff
@@ -39,32 +39,37 @@ -(NSURL*) getURL
...
 
 -(id) initWithURL:(NSURL*) url
 {
     assert(url != nil);
-    assert(self = [super init]);
-    // MLUploadQueueItem* item = [MLUploadQueueItem alloc];
-    _type = UPLOAD_QUEUE_TYPE_URL;
-    self.url = url;
-    self.image = nil;
+    if(self = [super init])
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MLUploadQueueItem.m]: and again

File: Monal/Classes/MLUploadQueueItem.m (hunk 1 above)
Comment (tmolitor-stud-tu) [reply Monal/Classes/MLUploadQueueItem.m]: [super init] will always return a non-nil value, no need to check it...

**Review by tmolitor-stud-tu — CHANGES_REQUESTED** [review]
no

File: Monal/Classes/MLUploadQueueItem.m (hunk 1 above)
Comment (janst97) [reply Monal/Classes/MLUploadQueueItem.m]: I used this as reference: https://developer.apple.com/documentation/objectivec/nsobject/1418641-init?language=objc

"If the new object can’t be initialized, the method should return nil. For example, a hypothetical BuiltInCamera class might return nil from its init method if run on a device that has no camera.". However, this does not apply to NSObject, so it's not necessary here. I'll fix it.

**Review by tmolitor-stud-tu — APPROVED** [review]

**Pull request merged**